### ISSUE TITLE
native events calibration errors

### ISSUE THREAD

**Issue by mfairchild365** [AmadeusITGroup/Assistive-Webdriver#526]
The first time I set up the project, I used the webdriver implementation (rather than playwright). I kept getting this error:

```
WebDriverError: Calibration for native events failed: Error: could not find the 974x645 rectangle filled with color [255,0,0,255] (tolerance 50), screenshot was not saved.
```

My apparent fix was to run standalone webdriver server in the VM at elevated permissions. I'm not confident that actually fixed the problem... maybe it was a coincidence?

Additionally, the custom calibration is very visual. A browser opens and displays a red triangle, and a screenshot is taken to find the triangle and perform the calibration (I acknowledge that I might be oversimplifying things). Because of this, nothing can overlap the rectangle (not another window or anything). The NVDA speech viewer and potentially, NVDA's visual outline can also cause this this error.

I'm wondering if there is a way to mitigate this?

**Comment by divdavem** [comment]
Hello @mfairchild365 
Thank you for your interest in Assistive-Webdriver.

> My apparent fix was to run standalone webdriver server in the VM at elevated permissions. I'm not confident that actually fixed the problem... maybe it was a coincidence?

I cannot explain why this would fix the problem. Note that it is possible to use the `--failed-calibrations-folder` option when starting assistive-webdriver to enable the option to save a screenshot when the calibration fails, in order to help debugging why it failed.

> Additionally, the custom calibration is very visual. A browser opens and displays a red triangle, and a screenshot is taken to find the triangle and perform the calibration (I acknowledge that I might be oversimplifying things). Because of this, nothing can overlap the rectangle (not another window or anything). The NVDA speech viewer and potentially, NVDA's visual outline can also cause this this error.

Note that with the latest version from the master branch of this repository (I will do a new release including those changes soon), the red rectangle is replaced with multiple QR codes and the detection of a single one is sufficient to know the position of the viewport, which makes calibration more reliable. In addition, during calibration, a click is done on the detected QR code to make sure the browser windows has the focus.

**Comment by divdavem** [comment]
The new release is now available, you can use [version 0.1.0 of assistive-webdriver](https://www.npmjs.com/package/assistive-webdriver/v/0.1.0) to benefit from the new QR code based calibration.
Note that the same mechanism is also used in [version 0.1.0 of the playwright version](https://www.npmjs.com/package/assistive-playwright-client/v/0.1.0).

**Comment by mfairchild365** [comment]
@divdavem thank you! It appears to be more reliable now. I'll close this issue and re-open if I run into further problems.
